# Naive Bayes

1. Compute and show the conditional probability distribution for each feature. Explain how you got these values and show your work. Note: You are expected to do this part of the question by hand. Explain how you got the probability distribution for at least two features in detail

In [ ]:
import pandas as pd

file = "Asssignment4_Data.xlsx"
sheet = "Train"

df = pd.read_excel(file)
df = df.drop(columns=["House ID"])

target = "Construction type"
features = [col for col in df.columns if col != target]

# class priors P(Y)
print("Class priors:\n")
print(df[target].value_counts(normalize=True))
print()

# conditional distributions for each feature:
# mean and variance within each class
for feature in features:
    print("=" * 50)
    print("Feature:", feature)

    grouped = df.groupby(target)[feature]

    means = grouped.mean()
    variances = grouped.var(ddof=0)   # population variance for Gaussian NB

    print("\nMeans:")
    print(means)

    print("\nVariances:")
    print(variances)
    print()

Class priors:

Construction type
Apartment    0.35
House        0.35
Condo        0.30
Name: proportion, dtype: float64

Feature: Local Price

Means:
Construction type
Apartment    7.332743
Condo        7.415900
House        5.760743
Name: Local Price, dtype: float64

Variances:
Construction type
Apartment    11.207517
Condo        17.719672
House         0.278604
Name: Local Price, dtype: float64

Feature: Bathrooms

Means:
Construction type
Apartment    1.285714
Condo        1.333333
House        1.071429
Name: Bathrooms, dtype: float64

Variances:
Construction type
Apartment    0.275510
Condo        0.305556
House        0.030612
Name: Bathrooms, dtype: float64

Feature: Land Area

Means:
Construction type
Apartment    6.103857
Condo        6.024667
House        6.630900
Name: Land Area, dtype: float64

Variances:
Construction type
Apartment    9.101044
Condo        5.396577
House        4.335326
Name: Land Area, dtype: float64

Feature: Living area

Means:
Construction type
Apartme

The class priors are:
$$
P(\text{Apartment}) = 0.35
$$

$$
P(\text{House}) = 0.35
$$

$$
P(\text{Condo}) = 0.30
$$

I then computed the mean and variance of both local price and land area given each class. To do this, I used the following equations


The mean is

$$
\mu = \frac{1}{n}\sum_{i=1}^{n} x_i
$$

and the variance is

$$
\sigma^2 = \frac{1}{n}\sum_{i=1}^{n}(x_i-\mu)^2
$$

The local price values are:
$$
\text{Local Price} \mid \text{Apartment} \sim \mathcal{N}(7.3,\ 11.2)
$$

$$
\text{Local Price} \mid \text{House} \sim \mathcal{N}(5.8,\ 0.3)
$$

$$
\text{Local Price} \mid \text{Condo} \sim \mathcal{N}(7.4,\ 17.7)
$$

The land area values are:

# Decision Tree